src/containers/sound-editor.jsx

```diff
@@ -51,7 +59,11 @@ class SoundEditor extends React.Component {
             trimEnd: null
         });
     }
-    submitNewSamples (samples, sampleRate) {
+    submitNewSamples (samples, sampleRate, skipUndo) {
+        if (!skipUndo) {
```

Comment:
Maybe it would be worthwhile to break out undo/redo to be independent of audio, so it could be tested by itself and maybe used in other places (then again if sound editor is going to be its own repo it wouldn't really be sharable)